lib/active_admin/filters/resource_extension.rb

```diff
@@ -128,13 +128,30 @@ def default_association_filters
           # remove deeply nested associations
           not_poly.reject! { |r| r.chain.length > 2 }
 
-          filters = poly.map(&:foreign_type) + not_poly.map(&:name)
+          # Check high-arity associations for filterable columns
+          high_arity, low_arity = not_poly.partition { |r| r.klass.count > namespace.maximum_association_filter_arity }
```

Comment:
Counting records has potential to impact performance in unexpected ways. There should also be an option to skip this check altogether. Another aspect of the performance impact is a additional query per association.

I'd also consider this as a major change, and require a major version bump. Is there about upgrading doc to add this new configuration to?

Reply:
@zorab47  agree this should not be a default behaviour

Reply:
Another option would be to ask the database for an approximate count from its internal statistics. That should be good enough for this use case (it's not 100% accurate, but querying for the approximate count is effectively free).

Postgres has [`reltuples`](https://wiki.postgresql.org/wiki/Count_estimate) and MySQL has [`table_rows`](https://dba.stackexchange.com/a/151790). Does ActiveAdmin support any other databases?

Reply:
If the check is being skipped / it's not the default behavior, does it still require a major version bump?